### PR TITLE
Add star_align_reads component

### PR DESCRIPTION
**Pull request by rcannood**
## Issue ticket number

Closes #xxxx 

## Checklist before requesting a review

- [x] I have performed a self-review of my code

- [x] Conforms to the [Contributing guidelines](https://github.com/viash-hub/base/blob/main/CONTRIBUTING.md)

- [x] Proposed changes are described in the [CHANGELOG.md](https://github.com/viash-hub/base/blob/main/CHANGELOG.md)

- [x] I have tested my code with `viash ns test --parallel -q <name or namespace>`

- Check the correct box. Does this PR contain:
  - [ ] Breaking changes
  - [x] New functionality
  - [ ] Major changes
  - [ ] Minor changes
  - [ ] Documentation
  - [ ] Bug fixes

### CODE REVIEW

**Review by KaiWaldrant — APPROVED** [review]
LGTM!

File: src/star/star_align_reads/script.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    "chimeric_junctions": "Chimeric.out.junction",
+    "log": "Log.final.out",
+    "splice_junctions": "SJ.out.tab",
+    "unmapped": "Unmapped.out.mate1",
+    "unmapped_r2": "Unmapped.out.mate2"
+}
+output_paths = {name: par[name] for name in expected_outputs.keys()}
+for name in expected_outputs.keys():
+    par[name] = None
+
```
Comment (DriesSchaumont) [inline src/star/star_align_reads/script.py:66]: If these keys are not needed anymore:
```suggestion
output_paths = {name: par.pop(name) for name in expected_outputs.keys()}
```

Comment (rcannood) [reply src/star/star_align_reads/script.py:66]: This makes it less clear to me what is happening. If I look at a dictionary expression like this, I don't expect it to have side-effects on `par` without really reading the code.

Comment (DriesSchaumont) [reply src/star/star_align_reads/script.py:66]: Sure, I can understand that

**Review by DriesSchaumont — APPROVED** [review]
LGTM!

**Pull request merged**